agent/consul/leader_peering.go

```diff
@@ -30,8 +31,13 @@ import (
 	"github.com/hashicorp/consul/proto/pbpeerstream"
 )
 
+var leaderExportedServicesCountKeyDeprecated = []string{"consul", "peering", "exported_services"}
 var leaderExportedServicesCountKey = []string{"consul", "peering", "exported_services"}
```

Comment:
should line 33 be without the `consul` prefix then ?

Also 🤦🏼 since I added that code line and forgot about the double prefixing issue.

Again, not sure if this is widely known so please bear with me as I brain dump something about prefixing these w the service name, ie `consul`.

This is mostly a Promethues thing, so this func runs over all metrics (gauges, counters, summaries) and prepends the service name: https://github.com/hashicorp/consul/blob/main/agent/setup.go#L192-L347 

But that only works if they are predeclared somewhere like `LeaderPeeringMetrics` below.

So it may be possible that some of these metrics are never predeclared (not the peering metrics here but some of the older ones). In that case, emitting the `consul` prefix might not be ideal. I've not spent time looking at which metrics don't actually get predeclared so I am just offering this as a warning -- apologies in advance if it's a wild goose chase instead.

Reply:
thanks for that catch! it looks like all of the metrics updated in this PR are predeclared.  But enforcing that should be a follow up to this.